src/pagination/get_data.jsx

```diff
@@ -0,0 +1,544 @@
+        text: 'Like',
+    },
+]
+
+const filterIcons = [
+    {
+        path: 'filter.svg',
+        alt: 'filter',
+        text: 'Filter',
+    },
+    {
+        path: 'quadro_sircle.svg',
+        alt: 'quadro_sircle',
+        text: ''
+    },
+    {
+        path: 'veiew_list.svg',
+        alt: 'veiew_list',
+        text: '',
+    },
+]
+
+const filterIconList = filterIcons.map(icon => {
+    return (
+        <div className='filter_icon_list_item'>
+            <img src={`/photos/${icon.path}`} alt={icon.alt} />
+            {icon.text && <p>{icon.text}</p>}
+        </div>
+    )
+})
+
+const productHooverItemsList = productHooverItems.map(item => {
+    return (
+        <Link to='' className='product_link_item'>
+            <img src={`/photos/products/${item.path}`} alt={item.text} />
+            <Link to='' className='product_link_item_text'>{item.text}</Link>
+        </Link>
+    )
+})
+
+const ProductList = ({ products }) => {
+    const navigate = useNavigate();
+
+    return products.map((product) => {
+        const handleClick = () => {
+            const newPath = `/products/${product.name}`;
+            navigate(newPath);
+        };
+
+        return (
+            <div className='product_item_shop' onClick={handleClick}>
+                <div className='item_hoover_state'>
+                    <button className='item_button'>Add to cart</button>
+                    <div className='product_link_items'>{productHooverItemsList}</div>
+                </div>
+                <Link to={`/products/${product.name}`} className='product_link'>
+                    <img src={`/photos/products/${product.path}`} alt={product.name} />
+                    {product.discount && <div className='item_circle item_discount'>{product.discount}</div>}
+                    {product.new && <div className='item_circle item_new'>New</div>}
+                    <div className='item_text'>
+                        <h3 className='item_name'>{product.name}</h3>
+                        <p className='item_type'>{product.type}</p>
+                        <div className='item_cost'>
+                            <p className='current_cost'>Rp {product.cost}</p>
+                            {product.oldCost && <p className='old_cost'> Rp {product.oldCost}</p>}
+                        </div>
+                    </div>
+                </Link>
+            </div>
+        )
+    })
+}
+
+
+const PaginatedItems = () => {
+    const [itemsPerPage, setItemsPerPage] = useState('16');
+    const [pageCount, setPageCount] = useState(0);
+    const [itemOffset, setItemOffset] = useState(0);
+    const [isLoading, setIsLoading] = useState({
+        filterArrVal: true,
+        currentPageVal: false,
+    });
+    const [filterOn, setFilterOn] = useState('default');
+    const [currentPage, setCurrentPage] = useState(JSON.parse(localStorage.getItem('localCurrentPage')) || 0);
+    const [filteredArray, setFilteredArray] = useState(allproducts)
```

Comment:
what does it mean `filteredArray`? variables should always describe what value they hold. do not use generic variable names